### PR TITLE
Touch Bug #1416256: Add tag inlineformset for project tags

### PR DESCRIPTION
**Pull request by phlax**
Allows adding and removing tags from a project

### CODE REVIEW

**Comment by mathjazz** [comment]
This patch has bugs, let's fix them before reviewing the code:

1. The order of slug and name field needs to be flipped.

1. The slug field should autocomplete on blur on the name field, lake it does for the project name and slug.

1. If you don't provide a `slug` for a tag, the error message has a 5px padding on the left, which is not needed.

1. If you don't provide a `name` for a tag, the error message ("This field is required.") is displayed under the `slug` field.

1. Priority can be set to a negative value and value above 5. If you save, you get and error, but you don't see where in the form it happened. I suggest we use the same dropdown as we use for project priority.

1. If you "Add more" tags, only the labels appear, field don't.

1. If you then try to save, you get a server error.
```
Traceback:

File "/Users/mathjazz/Desktop/pontoon/venv/lib/python2.7/site-packages/django/core/handlers/exception.py" in inner
  41.             response = get_response(request)

File "/Users/mathjazz/Desktop/pontoon/venv/lib/python2.7/site-packages/django/core/handlers/base.py" in _legacy_get_response
  249.             response = self._get_response(request)

File "/Users/mathjazz/Desktop/pontoon/venv/lib/python2.7/site-packages/django/core/handlers/base.py" in _get_response
  187.                 response = self.process_exception_by_middleware(e, request)

File "/Users/mathjazz/Desktop/pontoon/venv/lib/python2.7/site-packages/django/core/handlers/base.py" in _get_response
  185.                 response = wrapped_callback(request, *callback_args, **callback_kwargs)

File "/Users/mathjazz/Desktop/pontoon/venv/lib/python2.7/site-packages/django/utils/decorators.py" in inner
  185.                     return func(*args, **kwargs)

File "/Users/mathjazz/Desktop/pontoon/pontoon/administration/views.py" in manage_project
  145.                 and (tag_formset.is_valid() if tag_formset else True))

File "/Users/mathjazz/Desktop/pontoon/venv/lib/python2.7/site-packages/django/forms/formsets.py" in is_valid
  321.         self.errors

File "/Users/mathjazz/Desktop/pontoon/venv/lib/python2.7/site-packages/django/forms/formsets.py" in errors
  295.             self.full_clean()

File "/Users/mathjazz/Desktop/pontoon/venv/lib/python2.7/site-packages/django/forms/formsets.py" in full_clean
  343.         for i in range(0, self.total_form_count()):

File "/Users/mathjazz/Desktop/pontoon/venv/lib/python2.7/site-packages/django/forms/formsets.py" in total_form_count
  116.             return min(self.management_form.cleaned_data[TOTAL_FORM_COUNT], self.absolute_max)

File "/Users/mathjazz/Desktop/pontoon/venv/lib/python2.7/site-packages/django/utils/functional.py" in __get__
  35.         res = instance.__dict__[self.name] = self.func(instance)

File "/Users/mathjazz/Desktop/pontoon/venv/lib/python2.7/site-packages/django/forms/formsets.py" in management_form
  98.                     code='missing_management_form',

Exception Type: ValidationError at /admin/projects/firefox/
Exception Value: [u'ManagementForm data is missing or has been tampered with']
```

**Comment by mathjazz** [comment]
Nice work!

I'd still like to see 2 things fixed:

1. Error message for Name and Slug field should be aligned left. screen shot 2018-03-16 at 11 08 45

2. The error message for the priority breaks into multiple lines. The dropdown will take us closer to a bullet-proof experience. If you'll need more space, feel free to make the Name field narrower.

**Comment by phlax** [comment]
> I'd still like to see 2 things fixed:

@mathjazz these should be fixed now

File: pontoon/administration/views.py [hunk 1]
```diff
@@ -196,11 +213,13 @@ def manage_project(request, slug=None, template='admin_project.html'):
         'form': form,
         'subpage_formset': subpage_formset,
         'repo_formset': repo_formset,
+        'tag_formset': tag_formset,
         'external_resource_formset': external_resource_formset,
         'locales_selected': locales_selected,
         'locales_available': Locale.objects.exclude(pk__in=locales_selected),
         'subtitle': subtitle,
         'pk': pk,
+        'project': project,
```
Comment (mathjazz) [inline pontoon/administration/views.py]: It seems like we don't need that.

File: pontoon/administration/templates/tag_inline_formset.html [hunk 2]
```diff
@@ -0,0 +1,62 @@
+{% macro render(title, type, formset, project) %}
```
Comment (mathjazz) [inline pontoon/administration/templates/tag_inline_formset.html]: The `project` param is never used.

File: pontoon/administration/views.py (hunk 1 above)
Comment (phlax) [reply pontoon/administration/views.py]: ill check and remove...

Comment (phlax) [reply pontoon/administration/views.py]: it turns out its there for later PR - leaving in as it will be used quite soon

**Pull request merged**